web/modules/custom/rdf_entity/src/RdfEntityIdPluginManager.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace Drupal\rdf_entity;
+
+use Drupal\Component\Plugin\FallbackPluginManagerInterface;
+use Drupal\Core\Cache\CacheBackendInterface;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Extension\ModuleHandlerInterface;
+use Drupal\Core\Plugin\DefaultPluginManager;
+use Drupal\rdf_entity\Annotation\RdfEntityId;
+
+/**
+ * Plugin manager for entity ID generator plugins.
+ */
+class RdfEntityIdPluginManager extends DefaultPluginManager implements FallbackPluginManagerInterface {
+
+  /**
+   * Cached plugin instances.
+   *
+   * @var \Drupal\rdf_entity\RdfEntityIdPluginInterface[][]
+   */
+  protected $instances = [];
+
+  /**
+   * The entity type manager service.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Constructs the RdfEntityIdPluginManager object.
+   *
+   * @param \Traversable $namespaces
+   *   An object that implements \Traversable which contains the root paths
+   *   keyed by the corresponding namespace to look for plugin implementations.
+   * @param \Drupal\Core\Cache\CacheBackendInterface $cache_backend
+   *   Cache backend instance to use.
+   * @param \Drupal\Core\Extension\ModuleHandlerInterface $module_handler
+   *   The module handler.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager service.
+   */
+  public function __construct(\Traversable $namespaces, CacheBackendInterface $cache_backend, ModuleHandlerInterface $module_handler, EntityTypeManagerInterface $entity_type_manager) {
+    parent::__construct('Plugin/rdf_entity/Id', $namespaces, $module_handler, RdfEntityIdPluginInterface::class, RdfEntityId::class);
+    $this->alterInfo('rdf_taxonomy_tid_info');
+    $this->setCacheBackend($cache_backend, 'rdf_taxonomy_tid_plugins');
+    $this->entityTypeManager = $entity_type_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFallbackPluginId($plugin_id, array $configuration = []) {
+    return 'fallback';
+  }
+
+  /**
+   * Initializes the proper plugin given a RDF entity.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity.
+   *
+   * @return \Drupal\rdf_entity\RdfEntityIdPluginInterface
+   *   The plugin.
+   */
+  public function getPlugin(EntityInterface $entity) {
+    $entity_type_id = $entity->getEntityTypeId();
```

Comment:
I think it would be a good idea. More practical from the user perspective. But what about entity types that don't have a UI for bundle edit? I guess we can store directly in config the 3rd party setting.